tests/python_package_test/test_plotting.py

```diff
@@ -21,6 +22,20 @@ def breast_cancer_split():
                             test_size=0.1, random_state=1)
 
 
+@pytest.fixture(scope="module")
+def categorical_data():
+    def _categorical_data(category_values_lower_bound, category_values_upper_bound):
+        X, y = load_breast_cancer(return_X_y=True)
+        X_df = pd.DataFrame()
+        rnd = np.random.RandomState(0)
+        n_cat_values = rnd.randint(category_values_lower_bound, category_values_upper_bound, size=X.shape[1])
+        for i in range(X.shape[1]):
+            bins = np.linspace(0, 1, num=n_cat_values[i]+1)
+            X_df[f"cat_col_{i}"] = pd.qcut(X[:, i], q=bins, labels=range(n_cat_values[i])).as_unordered()
+        return train_test_split(X_df, y, test_size=0.1, random_state=1)
```

Comment:
If the only test using this function is going to just train on whatever data you pass it, performing `train_test_split()` is unnecessary computation and complexity. Please just return `X_df` and `y` directly, to simplify this function.